### ISSUE TITLE
[Feature Request] click count - and sort by clicks nr's, date added, date last click

### ISSUE THREAD

**Issue by e-p-s** [hoarder-app/hoarder#107]
Dear Development Team,

I would like to propose an enhancement for our bookmark manager that I believe would significantly improve user experience and functionality. The feature request involves the implementation of click counters for each bookmark, along with additional sorting options and a toggle switch for different views in the list window.

### Feature Details:

**Click Counter Per Bookmark:**
Introduce a click counter feature for each bookmark in the manager.
The click counter should track the number of times each bookmark is clicked or accessed by the user.
This feature will provide users with valuable insights into the popularity and relevance of their saved bookmarks.
**Sorting Options:**
Enable sorting bookmarks based on:
Number of clicks (descending order)
Date added (oldest to newest or vice versa)
Date of last click (recently accessed to least recently accessed)
**Toggle Switch for View Options:**
Implement a toggle switch within the list window to allow users to switch between different views easily.
Views may include grid view, list view, or any other relevant display options.
This feature will enhance user customization and adaptability to different preferences.

**Rationale:**

The click counter feature will empower users to prioritize bookmarks based on their usage frequency, improving efficiency and organization.
Offering various sorting options will cater to diverse user preferences, enhancing the overall usability of the bookmark manager.
The addition of a view toggle switch will provide users with greater flexibility in managing their bookmarks according to their preferred visual presentation.
Overall, these enhancements will elevate the functionality and user experience of our bookmark manager, making it more intuitive and efficient for users to organize and access their saved bookmarks.

Thank you for considering this feature request. I look forward to seeing these improvements implemented in future updates.

**Comment by MohamedBassem** [comment]
Hi @e-p-s, first of all, thanks for taking the time writing this feature request!

> Toggle Switch for View Options:

This was actually introduced in the last release (0.12.0):

![image](https://github.com/MohamedBassem/hoarder-app/assets/2418637/01ec4ec9-8d8e-43e9-8e1f-9539990f53db)

> Sorting Options

That makes a lot of sense. I'll probably implement it at some time.

> Click Counter Per Bookmark

That one to be honest I'm not entirely sure of its usefulness. What does it mean to improve the efficiency of organization of a bookmark and how a click count would help? If you have some bookmarks that you know you'll access frequently, you can just add them to some quick access custom list for example.

**Comment by e-p-s** [comment]
You generate a dynamic best of list for all your links. For me i Hope i can replace Homepage,dashy, homar for this. 

I think it ist Not a big thing, ist it?

**Comment by MikeKMiller** [comment]
I have to agree with Mohamed in not seeing the overall value in click counts. As these links are not intended to be public, what is the point in counting clicks?

**Comment by e-p-s** [comment]
argument for adding a click count and sorting feature in the Hoarder app, consider the following points:

1. **Automatic Popularity List**: Implementing a click counter would allow users to automatically generate a list of their most popular links, simplifying access to frequently visited sites.

2. **Enhanced User Insight**: Tracking clicks provides insights into user behavior, helping them understand which links they value most.

3. **Efficient Bookmark Management**: Users can organize bookmarks more effectively by prioritizing those with higher click counts, improving overall productivity.

4. **Personalized Recommendations**: Click data can be used to offer personalized link suggestions, enhancing the user experience by making relevant content more accessible. 

5. **Historical Usage Analysis**: Users can review historical click data to recall past interests or trends in their browsing habits.

6. **Minimal Programming Effort**: Implementing a click counter and sorting mechanism would not require significant programming effort, as it involves straightforward data tracking and basic sorting algorithms.

**Comment by MohamedBassem** [comment]
Sorting by date is coming in the next release. I don't think we'll do click counts though. So closing this as completed.

**Comment by ben-kenney** [comment]
I know this is closed but +1 for click counts. When you have lots of bookmarks it's useful to be able to quickly see your most frequently used bookmarks.

**Comment by thiswillbeyourgithub** [comment]
Similarly as @ben-kenney , after using karakeep for a while I am seriously missing that feature. It seems  to be a not overly complicated backend change but would allow:
- suggesting recently opened bookmarks (!)
- quickly resuming what you were reading on another device (!)
- quickly jump back on forth between two bookmarks when on mobile (in the browser you can open multiple tabs but not in the app)
- sorting by popularity

@MikeKMiller , after using for some time karakeep what do you think of this?

@MohamedBassem can you reconsider this FR? I am really surprised it didn't at least make it to the icebox. Even though I'd put it in the medium/low bin especially for the "open recent" feature.
